### DISCUSSION TITLE
Fetch observations by session id

### DISCUSSION THREAD

**Discussion by talrejanikhil** [langfuse/langfuse#4501] [Ideas]
Is there a way to fetch all observations by session id?
Right now the only way to do this is going through each trace and then iterating over every trace to reach observations:
```

traces = langfuse_client.fetch_traces(limit=limit, page=i, session_id=session_id).data
for t in traces:
    for o in t.observations:
        // do something
```

**Comment by jannikmaierhoefer** [comment]
Hi @talrejanikhil, fetching sessions is currently not supported by the low-level SDK. One option would be to fetch the session and the associated trace_ids via the /api/public/sessions/ endpoint of our API. Let me know if this would be an option for you.

**Reply by talrejanikhil** [reply]
Unfortunately this is not a great option because I still have to traverse each trace -> observation which is basically a double for loop. It would be nice at some point to have an API that returns observations by session id

**Reply by marcklingen** [reply]
can you open an idea thread for this? agree that this would be nice!

**Reply by talrejanikhil** [reply]
I have converted this one to an idea. Hope that is ok
